### ISSUE TITLE
CookieError Illegal key with CILogon auth

### ISSUE THREAD

**Issue by zonca** [jupyterhub/oauthenticator#132]
hi,
I'm using Jupyterhub 0.8 and `oauthenticator` 0.7.1.

With CILogon, the usernames are URLs, it looks like this is not accepted as a cookie key so the single-user server throws an exception.

Any suggestion on a workaround? See traceback below
```

    Traceback (most recent call last):
File "/opt/conda/lib/python3.6/site-packages/tornado/web.py", line 1509, in _execute                                          
        result = method(*self.path_args, **self.path_kwargs)                                                                              File "/opt/conda/lib/python3.6/site-packages/tornado/web.py", line 2886, in wrapper                                           
        url = self.get_login_url()                                                                                                        
File "/opt/conda/lib/python3.6/site-packages/jupyterhub/services/auth.py", line 690, in get_login_url                         
        state = self.hub_auth.set_state_cookie(self, next_url=self.request.uri)                                                           File "/opt/conda/lib/python3.6/site-packages/jupyterhub/services/auth.py", line 562, in set_state_cookie                      
        **kwargs                                                                                                                          
File "/opt/conda/lib/python3.6/site-packages/tornado/web.py", line 619, in set_secure_cookie                                  
        expires_days=expires_days, **kwargs)                                                                                              
File "/opt/conda/lib/python3.6/site-packages/tornado/web.py", line 548, in set_cookie                                         
        self._new_cookie[name] = value                                                                                                    
File "/opt/conda/lib/python3.6/http/cookies.py", line 521, in __setitem__                                                     
        self.__set(key, rval, cval)                                                                                                       
File "/opt/conda/lib/python3.6/http/cookies.py", line 511, in __set                                                           
        M.set(key, real_value, coded_value)                                                                                               
File "/opt/conda/lib/python3.6/http/cookies.py", line 380, in set
        raise CookieError('Illegal key %r' % (key,))                                                                                    
http.cookies.CookieError: Illegal key 'user-http%3A//cilogon.org/servera/users/12345-oauth-state'

```

**Comment by minrk** [comment]
This is related to #133. We shouldn't be using these URLs as usernames.

**Comment by minrk** [comment]
closed by #134
